daemon/__init__.py

```diff
@@ -0,0 +1,80 @@
+import json
+import subprocess
+import threading
+from collections import namedtuple
+
+import pkg_resources
+from fastapi import FastAPI
+from jina.logging import JinaLogger
+from uvicorn import Config, Server
+
+daemon_logger = JinaLogger(context='👻 JINAD')
+
+
+def _get_app():
+    from .api.endpoints import common_router, flow, pod, pea, logs
+    from .config import jinad_config, fastapi_config, server_config, openapitags_config
+
+    context = namedtuple('context', ['router', 'openapi_tags', 'tags'])
+    _all_routers = {
+        'flow': context(router=flow.router,
+                        openapi_tags=openapitags_config.FLOW_API_TAGS,
+                        tags=[openapitags_config.FLOW_API_TAGS[0]['name']]),
+        'pod': context(router=pod.router,
+                       openapi_tags=openapitags_config.POD_API_TAGS,
+                       tags=[openapitags_config.POD_API_TAGS[0]['name']]),
+        'pea': context(router=pea.router,
+                       openapi_tags=openapitags_config.PEA_API_TAGS,
+                       tags=[openapitags_config.PEA_API_TAGS[0]['name']])
+    }
+    app = FastAPI(
+        title=fastapi_config.NAME,
+        description=fastapi_config.DESCRIPTION,
+        version=fastapi_config.VERSION
+    )
+    app.include_router(router=common_router,
+                       prefix=fastapi_config.PREFIX)
+    app.include_router(router=logs.router,
+                       prefix=fastapi_config.PREFIX)
+    if jinad_config.CONTEXT == 'all':
+        for _current_router in _all_routers.values():
+            app.include_router(router=_current_router.router,
+                               tags=_current_router.tags,
+                               prefix=fastapi_config.PREFIX)
+    else:
+        _current_router = _all_routers[jinad_config.CONTEXT]
+        app.openapi_tags = _current_router.openapi_tags
+        app.include_router(router=_current_router.router,
+                           tags=_current_router.tags,
+                           prefix=fastapi_config.PREFIX)
+    return app
+
+
+def write_openapi_schema(filename='schema.json'):
+    app = _get_app()
+    schema = app.openapi()
+    with open(filename, 'w') as f:
+        json.dump(schema, f)
+
+
+def _start_uvicorn(app: 'FastAPI'):
+    from .config import server_config
+    config = Config(app=app,
+                    host=server_config.HOST,
+                    port=server_config.PORT,
+                    loop='uvloop',
+                    log_level='error')
+    server = Server(config=config)
+    server.run()
+    daemon_logger.info('bye!')
+
+
+def _start_fluentd():
+    daemon_logger.info('starting fluentd')
+    cfg = pkg_resources.resource_filename('jina', 'resources/fluent.conf')
+    subprocess.Popen(['fluentd', '-c', cfg])
```

Comment:
Fluentd also ships as a daemon (td-agent), which means we can have systemctl manage it along with jinad. We can avoid having `fluentd` invocation inside python code, as this makes `jinad` unusable without `fluentd` command. 
https://github.com/jina-ai/jinad/blob/main/scripts/deb-systemd.sh#L49

Reply:
let's handle this as another PR. right now it is too ahead to put `systemctl` together, the goal of this PR is to make the jinad usable in CICD.

Reply:
btw,
```bash
curl -L https://raw.githubusercontent.com/jina-ai/jinad/main/scripts/deb-systemd.sh | bash
```
this is good, but later will make it shorter, something like `curl -L https://daemon-sh.jina.ai | bash`